CHANGELOG.md

```diff
@@ -0,0 +1,722 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/)
+and this project adheres to [Semantic Versioning](http://semver.org/).
+
+## Unreleased
+
+<!-- towncrier release notes start -->
+
+## 5.4.0 -- 2024-11-18
+
+### Core
+
+- Removed the `OIDC_AUDIENCE` setting
+
+### Agent
+
+- Changed auto-update task to reuse current scheduler instead of creating a new one
+- Fixed environment variables from the machine running the agent propagating to slurm jobs (notice `--export=ALL` is the default behavior for sbatch)
+- Removed the `OIDC_AUDIENCE` setting
+
+### API
+
+- Added clone capability to job submissions so they can be resubmitted to the cluster when needed [PENG-1676, ASP-4597]
+- Modernized Dockerfile to use multi-stage builds ([ASP-4598](https://jira.scania.com/browse/ASP-4598))
+- Bumped dependencies (FastAPI and fastapi-pagination)
+- Silenced logging in the `/health` endpoint
+- Added endpoints to upload by URL [([PENG-2426](https://app.clickup.com/t/18022949/PENG-2426))(https://sharing.clickup.com/t/h/c/18022949/PENG-2426/PYUL64MXH5P0MWB)]
+  - For Job Script files
+  - For Job Script Template files
+  - For Job Script Template workflow files
+
+### CLI
+
+- Added clone command to job submissions so they can be resubmitted to the cluster when needed [PENG-1677, ASP-4597]
+- Enabled positional arguments to select entries using the cli ([ASP-5649](https://jira.scania.com/browse/ASP-5649))
+- Modified internal details to ensure commands that require no authentication face no configuration errors ([ASP-5745](https://jira.scania.com/browse/ASP-5745))
+- Removed the `OIDC_AUDIENCE` setting
+
+## 5.3.0 -- 2024-09-09
+
+### Core
+
+- Refined authentication handler to get it ready to be used on the CLI and enable auto-login ([ASP-4779](https://jira.scania.com/browse/ASP-4779))
+  - Created a request handler to facilitate the communication with OIDC
+  - Added Pydantic back as a dependency and applied it to validate responses from the OIDC server
+  - Created extra exceptions to handle authentication errors, that in turn can be more easily handled by client code
+
+### Agent
+
+- Fixed issue when downloading job script files for job submission with large names
+- Cache slurm submissions to avoid the resubmission of the same job if the job status update fails ([PENG-2342](https://app.clickup.com/t/18022949/PENG-2342))
+
+### API
+
+- Fixed issue preventing the creation of a job script from an empty template file
+- Enhance logging for file storage operations
+- Implemented an endpoint to fetch a cluster status by its client id [([PENG-2348](https://app.clickup.com/t/18022949/PENG-2348))(https://sharing.clickup.com/t/h/c/18022949/PENG-2348/QWZFBKV72ZNL293)]
```

Comment:
Review the links. Some were already hyperlinks, so they got duplicated.

Reply:
solved by [539cad9](https://github.com/omnivector-solutions/jobbergate/pull/690/commits/539cad9b86d6c8a2ada5016b9f1a25111f43d027).